### ISSUE TITLE
적은 데이터를 보안하기 위한 방법 Augmentation

### ISSUE THREAD

**Issue by Namhyunho** [suimn416/2019.Spring.AI_Leader#21]
Augmentation  in kerase
[참고한 자료](https://keraskorea.github.io/posts/2018-10-24-little_data_powerful_model/)

**Comment by Namhyunho** [comment]
Augmentation in Pytorch
[참고한 자료](https://colab.research.google.com/drive/109vu3F1LTzD1gdVV6cho9fKGx7lzbFll)

**Comment by Namhyunho** [comment]
[코랩링크](https://github.com/suimn416/2019.Spring.AI_Leader/blob/master/%EC%9D%B4%EB%AF%B8%EC%A7%80_%EC%A0%84%EC%B2%98%EB%A6%AC(Augmentation).ipynb)

**Comment by socome** [comment]
@Namhyunho 해당 코드를 통해서 데이터를 가공해서 제공해주세욥

**Comment by Namhyunho** [comment]
참고할 Augmentation class  

데이터 전처리와 Augmentation
모델이 적은 이미지에서 최대한 많은 정보를 뽑아내서 학습할 수 있도록 우선 이미지를 augment 시켜보도록 하겠습니다. 이미지를 사용할 때마다 임의로 변형을 가함으로써 마치 훨씬 더 많은 이미지를 보고 공부하는 것과 같은 학습 효과를 내는 겁니다. 이를 통해 과적합 (overfitting), 즉 모델이 학습 데이터에만 맞춰지는 것을 방지하고, 새로운 이미지도 잘 분류할 수 있게 합니다.


@Junuu : [TORCHVISION.TRANSFORMS](https://pytorch.org/docs/stable/torchvision/transforms.html#torchvision.transforms.Normalize), 

@Namhyunho : [imgaug](https://github.com/aleju/imgaug)

추가적으로 시행할 상황들 각자 colab에서 작업후 계선된 3개의 class를 선정 하여  TORCHVISION과 imgaug 합쳐가면서 추가적인 작업을 진행 후 공용 개정에 추가할 계획입니다.

**Comment by Namhyunho** [comment]
진행 방향 변경 

준우와 현호 모두 imgang로 작업을 해본후 우리의 코드에 적용해보고 만약 작업이 안될시 차선으로  TORCHVISION으로 할예정

**Comment by Junuu** [comment]
[image agumentation의 개념](https://m.blog.naver.com/PostView.nhn?blogId=4u_olion&logNo=221437862590&categoryNo=50&proxyReferer=https%3A%2F%2Fwww.google.com%2F)

**Comment by Junuu** [comment]
[image agumentation 예시(colab code link)](https://colab.research.google.com/drive/1aPZ43V-MSg_PtgbDngBmEhn1c5EnohIX)

**Comment by Junuu** [comment]
[image weather agumentation document](https://imgaug.readthedocs.io/en/latest/source/api_augmenters_weather.html)

**Comment by Namhyunho** [comment]
### 데이터에 image agumentation 적용하기
 - 현재 train 셋에 데이터를 적용하기위해 이미지(PIL Image)를 NumPy배열 행태로 변환해야함.
 - torchvision.transforms.Compose또는 ImgAugTransform를 사용하여 agumentation 을 실행함.
 - torchvision.transforms.ToTensor CLASS를 통해 tensor형태로 변환후 우리 train에 적용 

[참고자료1](https://pytorch.org/docs/stable/torchvision/transforms.html#torchvision-transforms)
